### DISCUSSION TITLE
Can Sling do the T part of ELT?

### DISCUSSION THREAD

**Discussion by alberttwong** [slingdata-io/sling-cli#148] [Ideas]
It doesn't quite say it in the docs but I'm curious.

**Comment by flarco** [comment]
Sling only does the following transformations:
- Column type coercion, the `columns` key in source.options (https://docs.slingdata.io/sling-cli/run/configuration#source) 
- Special string transforms, like encodings, uuid parsing. See https://github.com/slingdata-io/sling-cli/blob/main/core/sling/tranforms.go
- JSON Flattenning: the `flatten` key in source.options (https://docs.slingdata.io/sling-cli/run/configuration#source) 
- [JMESPath](https://jmespath.org/) transformations for JSON source files: the `jmespath` key in source.options (https://docs.slingdata.io/sling-cli/run/configuration#source) 
- Custom SELECT query as the source stream.

There are no plans to supports other custom transformations.

**Reply by alberttwong** [reply]
We really should publish this info on the docs page.

**Reply by flarco** [reply]
Will do!

**Reply by flarco** [reply]
Added with examples 👉  https://docs.slingdata.io/sling-cli/run/transformations

**Comment by miguelgilmartinez** [comment]
Yes. I confirm I was able to export from a StarRocks table composed by 6 columns just the two of them to a MySQL blank schema. Sling created the new table and imported data properly. 

`./sling run --src-conn STARROCKS --src-stream "SELECT columnB, columnA FROM tbl WHERE columnB > 6000" -tgt-conn MYSQL --tgt-object=tbl --mode full-refresh`

In that proof of concept, Sling executed the SELECT and created a new table with right data types. Great job.

Note: Probably it is not a proper "T" part by Sling, but it can do the job.

**Reply by flarco** [reply]
True, added to the list
